contracts/modules/cover/Cover.sol

```diff
@@ -260,33 +261,39 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
           / (allocationRequest.previousExpiration - allocationRequest.previousStart); // previous period
       }
 
-      // converting asset amount to nxm and rounding up to the nearest NXM_PER_ALLOCATION_UNIT
-      uint coverAmountInNXM = Math.roundUp(
-        Math.divCeil(poolAllocationRequests[i].coverAmountInAsset * ONE_NXM, nxmPriceInCoverAsset),
-        NXM_PER_ALLOCATION_UNIT
-      );
-
-      uint premiumInNXM = stakingPool(poolAllocationRequests[i].poolId).requestAllocation(
-        coverAmountInNXM,
-        vars.previousPremiumInNXM,
-        allocationRequest
-      );
-
-      // omit deallocated pools from the segment
-      if (coverAmountInNXM != 0) {
-        coverSegmentAllocations[allocationRequest.coverId][segmentId].push(
-          PoolAllocation(
-            poolAllocationRequests[i].poolId,
-            coverAmountInNXM.toUint96(),
-            premiumInNXM.toUint96()
-          )
+      {
+        // converting asset amount to nxm and rounding up to the nearest NXM_PER_ALLOCATION_UNIT
+        uint coverAmountInNXM = Math.roundUp(
+          Math.divCeil(poolAllocationRequests[i].coverAmountInAsset * ONE_NXM, nxmPriceInCoverAsset),
+          NXM_PER_ALLOCATION_UNIT
+        );
+        if (poolAllocationRequests[i].allocationId == type(uint).max) {
+          allocationRequest.allocationId = type(uint).max;
+        } else {
+          allocationRequest.allocationId = poolAllocationRequests[i].allocationId;
+        }
```

Comment:
```
if (source == X) {
  target = X;
} else {
  target = source;
}
```
wouldn't target always get assigned the source value? :grin: 

Instead of trusting user input, this should be assigned based on whether there's a previous allocation id or not. This check is done at line 249 (`if (vars.previousPoolAllocationsLength > i)`).